test/spec.index.js

```diff
@@ -71,6 +71,150 @@ describe('Template Mixins', function () {
 
     });
 
+    describe('input-date', function () {
+
+        beforeEach(function () {
+            middleware = mixins(translate, {});
+        });
+
+        it('adds a function to res.locals', function () {
+            middleware(req, res, next);
+            res.locals['input-date'].should.be.a('function');
+        });
+
+        it('returns a function', function () {
+            middleware(req, res, next);
+            res.locals['input-date']().should.be.a('function');
+        });
+
+        it('renders thrice if the field is not marked as inexact', function () {
+            middleware(req, res, next);
+            res.locals['input-date']().call(res.locals, 'field-name');
+            render.should.have.been.calledThrice;
+        });
+
+        it('renders twice if the field is marked as inexact', function () {
+            var middlewareWithFieldNameMarkedAsInexact = mixins(translate, {
+                'field-name': {
+                    'inexact': true
+                }
+            });
+            middlewareWithFieldNameMarkedAsInexact(req, res, next);
+            res.locals['input-date']().call(res.locals, 'field-name');
+            render.should.have.been.calledTwice;
+        });
+
+        it('looks up field label', function () {
+            middleware(req, res, next);
+            res.locals['input-date']().call(res.locals, 'field-name');
+
+            render.called;
+
+            var dayCall = render.getCall(0),
+                monthCall = render.getCall(1),
+                yearCall = render.getCall(2);
+
+            dayCall.calledWith(sinon.match({
+              'class': 'date-input',
+              error: undefined,
+              hint: null,
+              id: 'field-name-day',
+              label: 'fields.field-name-day.label',
+              max: 31,
+              maxlength: 2,
+              min: 1,
+              pattern: '[0-9]*',
+              required: true,
+              type: 'text',
+              value: undefined
+            })).should.be.true;
```

Comment:
Couple of points of style here that will make the tests a little easier to understand:

* sinon-chai will give you `dayCall.should.have.been calledWith(...)`, which (IMO) is a little easier to parse.
* Using `sinon.match` means you don't need to specify the contents of the entire object, just the keys you're interested in for a particular test. So in this case, it should only check for the label - if you want to test other properties then write tests for those properties, don't use a side effect of this test.